src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskDelayScheduler.cs

```diff
@@ -76,9 +76,18 @@ private async Task ThrottleAsync(Func<CancellationToken, Task> asyncFnctionToCal
             try
             {
                 // First we wait the delay time. If another request has been made in the interval, then this task
-                // is cancelled. To avoid unnecessary OperationCanceled exceptions it tests to see if the token has
-                // been canceled
-                await Task.Delay(TaskDelayTime);
+                // is cancelled.
+                try
+                {
+                    if (!token.IsCancellationRequested)
+                    {
+                        await Task.Delay(TaskDelayTime, token);
```

Comment:
I wonder if this will make debugging unit tests more painful.  Still seems better though.

Reply:
I usually disable breaking on `*CanceledException`. It probably still slows down both the debugger and debuggee.

Reply:
Same, and exactly - it does slow them down some.